kof-operator/internal/k8s/pods.go

```diff
@@ -2,14 +2,15 @@ package k8s
 
 import (
 	"context"
+	"slices"
 
 	corev1 "k8s.io/api/core/v1"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-const PrometheusReceiverAnnotation = "k0rdent.mirantis.com/kof-prometheus-receiver"
+const CollectorMetricsAnnotation = "k0rdent.mirantis.com/kof-collector-metrics"
```

Comment:
Why do you use annotation rather than a label? Label would allow you to request all pods collectors by label and then you won't need to filter them by annotation. Request all pods might require pagination as the cluster might have a huge number of pods.